### ISSUE TITLE
Spotify - Stream appears to play but no sound output - HA media player stuck buffering.

### ISSUE THREAD

**Issue by jazzyisj** [music-assistant/hass-music-assistant#2982]
### What version of Music Assistant has the issue?

2.3.0b28

### What version of the Home Assistant Integration have you got installed?

2024.9.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Streaming Spotify - no sound output.    

MASS media player appears to be playing with album art, progress indicator etc.  HA media player remains in buffering state.  Youtube, TuneIn, Radio, and local files all work as expected.  

I've removed the spotify provider and reinstalled it.  I've cleared the core cache.  I've restarted the server and reloaded the integration.  Spotify plays from computers, phones, and on all cast devices without MASS with no issues.  Happens on all media players (Cast, Fully Kiosk)

<img width="1229" alt="image" src="https://github.com/user-attachments/assets/776631a3-92f0-4d40-914f-9244117d88b2">
<img width="1220" alt="image" src="https://github.com/user-attachments/assets/b947d5f5-adca-4730-b827-f2b4f8fd33e6">

### How to reproduce

Play a spotify playlist, album, or song.

### Music Providers

Spotify

### Player Providers

All media players.  Cast (Nest Hubs, Nest Mini's, Chromecasts, and Google Home), Fully Kiosk.

### Full log output

[music-assistant.log](https://github.com/user-attachments/files/17182456/music-assistant.log)

### Additional information

This has been happening fore the last couple beta releases.

### What version of Home Assistant Core are your running

2024.10.0b4

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

ODROID

**Comment by jazzyisj** [comment]
Yep, Public IP looks ok.  192.168.1.100

**Comment by marcelveldt** [comment]
Did you check another source, like radio station to verify if this is isolated to spotify ?

**Comment by jazzyisj** [comment]
Updated to  2.3.0b29, problem persists but now I'm getting errors in the MASS log.

[music-assistant.log](https://github.com/user-attachments/files/17197574/music-assistant.log)

**Comment by jazzyisj** [comment]
Yes I have Spotify premium.  I've been using Spotify with MASS pretty much since the day you released it.  It only stopped working maybe 2 or 3 updates ago.   Am I really the only one having an issue?  I don't think anything has changed on my that might have caused this.

**Comment by marcelveldt** [comment]
We had/have some nasty issues with (Spotify) playback lately but this is not one of them tbh so the issue seems to be isolated to your case so far. 

The only other thing that could be going on is that its account related but that is super far stretched.
Do you have discord ? Can you PM me there please ? I like to send you a test account to try out to rule out if its maybe account related.

**Comment by jazzyisj** [comment]
I PM'd you on discord.   I've logged into my Spotify account to check and it's definitely a Premium Family Plan in good standing.  There is only one account on there now so that can't be the issue.    Spotify is working correctly on all my other devices.

This is the service call I've been using in Dev Tools to test (with various valid Playlist names).   I've also tried with Artist, Track, and Album with the same results.  

```
action: mass.play_media
data:
  media_id: All Out 60s
  media_type: playlist
target:
  entity_id:
    - media_player.office_hub_2
```

**Comment by jazzyisj** [comment]
We have discovered the issue!  Thank you for pointing me in the right direction marcelveldt.

The Pi-Hole on my network randomly starting blocking requests to a domain spotify is calling.  I don't know if my adlist got updated or if Spotify made a change but if you run into this issue check if these domains are getting blocked on your Pi-Hole and whitelist them if they are.

```
audio-ak-spotify-com.akamaized.net
audio-akp-spotify-com.akamaized.net
```
